src/Language/PureScript/Pretty/JS.hs

```diff
@@ -58,9 +57,9 @@ literals = mkPattern' match'
     , return $ emit "}"
     ]
     where
-    objectPropertyToString :: (Emit gen) => Text -> gen
-    objectPropertyToString s | identNeedsEscaping s = string s
-                             | otherwise = emit s
+    objectPropertyToString :: (Emit gen) => PSString -> gen
+    objectPropertyToString s | identNeedsEscaping $ T.pack $ codePoints s = string s
```

Comment:
> As far as I can tell, there are no more than three things we might need to do with a PSString:

We also

* rewrite the program to use the string as a label
* derive identifier names from the string
* pretty-print PureScript code